### DISCUSSION TITLE
Creating TDD for Consume method - Suggestions?

### DISCUSSION THREAD

**Discussion by rhamses** [aio-libs/aiokafka#1035] [Q&A]
Hi everyone!

I'm studying aiokafka with the goal to rewrite a few parts of a project at work. To convince the rest of the team to adopt the library I'd like to write a test to simulate the load of data we receive. 

I'd like to create the mock incoming data, opened the consumer and then follow with the rest of the tasks we have but the consumer is blocking the execution. How it'd be the best way to test the consumer?

**Comment by ods** [comment]
Do you need to run producer concurrently along with consumer? Start it as a separate process or asyncio task.

**Comment by marcosschroh** [comment]
Just sharing: I have created a micro framework [kstreams](https://kpn.github.io/kstreams/) based on `aiokafka` which does some of the things that you want.
